lib/mongo_client.js

```diff
@@ -109,7 +109,7 @@ const validOptions = require('./operations/mongo_client_ops').validOptions;
  * @param {boolean} [options.auto_reconnect=true] Enable auto reconnecting for single server instances
  * @param {boolean} [options.monitorCommands=false] Enable command monitoring for this client
  * @param {number} [options.minSize] If present, the connection pool will be initialized with minSize connections, and will never dip below minSize connections
- * @param {boolean} [options.useNewUrlParser=false] Determines whether or not to use the new url parser
+ * @param {boolean} [options.useNewUrlParser=false] Determines whether or not to use the new url parser. If true, the new url parser will be used to verify the connection string, which will help ensure that connection strings work with the new url parsing. In the future, this option will be removed and the new url parser will always be used.
```

Comment:
I'm not sure this is actually accomplishing the request of the ticket. Mostly I'm concerned that we're recursively using the term "new url parser" in the very definition of the option. Could we say something like: "Enables the new, spec-compliant, url parser shipped in the core driver.  This url parser fixes a number of problems with the original parser, and aims to outright deprecate that parser in the near future."